lib/Dredd.js

```diff
@@ -31,153 +99,83 @@ class Dredd {
       end: 0,
       duration: 0,
     };
-    this.files = [];
-    this.transactions = [];
     this.transactionRunner = new TransactionRunner(this.configuration);
     this.logger = logger;
   }
 
-  run(callback) {
-    this.logger.debug('Configuring reporters');
-    configureReporters(this.configuration, this.stats, this.transactionRunner);
-
-    this.logger.debug('Resolving --require');
-    if (this.configuration.options.require) {
-      const requirePath = resolveModule(this.configuration.custom.cwd, this.configuration.options.require);
-      try {
-        // eslint-disable-next-line global-require, import/no-dynamic-require
-        require(requirePath);
-      } catch (err) {
-        this.logger.error(`Error requiring module '${requirePath}':`, err.message);
-        callback(err, this.stats);
-        return;
-      }
-    }
-
-    this.logger.debug('Resolving API descriptions locations');
+  prepareAPIdescriptions(callback) {
```

Comment:
The prepareAPIdescriptions() method isn't a real method of a real object as the Dredd class doesn't need to be a class in the first place. I created it just to make run() shorter. It roughly contains the pipeline turning API description locations into API descriptions, then parsing them, and then compiling them into transactions. Once we have async/await/promises and the code is less tied with 'this.configuration', it might make sense to refactor this further.